blog/arquillian-pact-1.0.0.Alpha2.textile

```diff
@@ -0,0 +1,73 @@
+---
+date: 2016-09-29
+author: lordofthejars
+---
+
+p. Arquillian Pact is an extension that provides an integration between Arquillian and "Pact":https://docs.pact.io/.
+
+h3 What's new
+
+p. <i class="icon-star"></i> Maven Loader
+
+p. Now you can use any Maven artifact to store contracts (pacts files). Then provider will resolve given artifact, uncompress it and execute the contracts found there.
+
+p. To use it you need to import dependency:
+
+bc(prettify).. <dependency>
+  <groupId>org.arquillian.pact</groupId>
+  <artifactId>arquillian-pact-provider-maven-loader</artifactId>
+</dependency>
+
+p. For example, this is how you can get contracts from a jar file stored at Maven repository. It is important to note that Arquillian Pact always takes the latest version of jar in case of not specifying one concrete version.
+
+bc(prettify).. @PactMavenDependency(value = "org.superbiz:contract:[1.0,]")
+public class MavenLoaderTest {}
+
+p. <i class="icon-star"></i> Git Loader
+
+p. Now you can get contracts from Git repository. The provider will clone/pull the repo and execute the contracts found there.
+
+p. To use it you need to import dependency:
+
+bc(prettify).. <dependency>
+  <groupId>org.arquillian.pact</groupId>
+  <artifactId>arquillian-pact-provider-git-loader</artifactId>
+</dependency>
+
+p. For example, this is how you can get contracts form a public Git repository:
+
+bc(prettify).. @PactGit("https://github.com/lordofthejars/starwarspact.git")
+public class GitLoaderTest {}
+
+p. Check __PactGit__ annotations to see the options for configuring authentication or set a concrete directory inside Git repo.
+
+p. <i class="icon-star"></i> Output Directory
+
+p. Now in consumer you can use __pactReportDir__ configuration property to set where contracts are generated.
+
+For example, this is how you can generate contracts into target/contracts directory.
```

Comment:
Well Pact generates the contract files from stub http definition. So although it sounds weird if we want to be strict Pact generates the contracts from code.